cpmpy/expressions/globalfunctions.py

```diff
@@ -444,6 +444,9 @@ def decompose_comparison(self, cmp_op, cpm_rhs):
         lbs, ubs = get_bounds(arr)
         lb, ub = min(lbs), max(ubs)
 
+        if lb == ub: # if the domain has only one value then check if it is n or not
```

Comment:
I think what you can do instead, is to make `bvars = boolvar(shape=(ub+1-lb,))` so it will always be an array.
We do smt similar in `make_assump_model` in tools.explain.utils.
I think that the original decomposition should work as is then.